src/app/applications/[id]/page.tsx

```diff
@@ -2,65 +2,82 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import PageContainer from "@/components/PageContainer";
+"use client";
+
+import Button from "@/components/Button";
 import PageHeading from "@/components/PageHeading";
 import Sidebar from "@/components/Sidebar";
-import Button from "@/components/button";
-import FieldAttachmentContainer from "@/components/fieldAttachmentContainer";
-import { faPaperPlane, faSave } from "@fortawesome/free-solid-svg-icons";
+import { useApplicationDetails } from "@/providers/application/ApplicationProvider";
+import { State } from "@/types/application.types";
+import {
+  formatApplicationState,
+  isApplicationComplete,
+} from "@/utils/application";
+import { faPaperPlane } from "@fortawesome/free-solid-svg-icons";
+import FormContainer from "./FormContainer";
 import { createApplicationSidebarItems } from "./sidebarItems";
 
-type ApplicationDetailsPageProps = {
-  params: { id: string };
-};
+export default function ApplicationDetailsPage() {
+  const { application, submitApplication } = useApplicationDetails();
+
+  if (!application) return;
+
+  const events = application.events;
+  const lastEvent = events[events.length - 1].eventType;
+  const sidebarItems = createApplicationSidebarItems(application);
 
-export default function ApplicationDetailsPage({
-  params,
-}: ApplicationDetailsPageProps) {
-  const { id } = params;
-  const sidebarItems = createApplicationSidebarItems();
   return (
-    <PageContainer>
-      <div className="mx-8 md:mx-auto md:w-3/4 xl:mx-0 xl:grid xl:w-full xl:grid-cols-12 xl:gap-x-20">
-        <div className="col-span-8 col-start-3 xl:col-span-6 xl:col-start-3">
-          <div className="px-3">
-            <div className="sm:flex sm:justify-between">
-              <PageHeading>Application {id}</PageHeading>
-              <div className="mt-4 flex gap-x-3 sm:mt-0">
-                <Button
-                  type="warning"
-                  text="Save changes"
-                  icon={faSave}
-                  className="h-fit text-[10px] sm:text-xs"
-                />
+    <div className="mx-8 mt-20 md:mx-auto md:w-3/4 xl:mx-0 xl:grid xl:w-full xl:grid-cols-12 xl:gap-x-20">
```

Comment:
instead of using custom class. Would you mind wrapping it inside PageContainer to keep consistency